docs/dense-retrieval.md

```diff
@@ -0,0 +1,35 @@
+# Direct PyPI Package Replication on Dense Retrieval
```

Comment:
Since we're not pushing out PyPI package, just call it "Dense Retrieval Replication".